lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -92,7 +92,9 @@ def group_name
 
     def run_kubectl(*args)
       raise FatalDeploymentError, "Namespace missing for namespaced command" if namespace.blank?
-      args = args.unshift("kubectl").push("--namespace=#{namespace}")
+      raise FatalDeploymentError, "Explicit context is required to run this command" if context.blank?
+      args = args.unshift("kubectl").push("--namespace=#{namespace}").push("--context=#{context}")
```

Comment:
The best way is to avoid mutation as much as possible, for example by:

```ruby
args = ["kubectl] + args + ["--namespace=#{namespace}", "--context=#{context}"]
```

In this case it creates a new array and doesn't mutate the original argument.